tracer/src/Datadog.Trace/Configuration/ConfigurationSources/Telemetry/ConfigurationBuilder.cs

```diff
@@ -493,23 +504,9 @@ private ConfigurationResult<T> GetAs<T>(Func<T, bool>? validator, Func<string, P
         /// <returns>The raw <see cref="ConfigurationResult{T}"/></returns>
         private ConfigurationResult<T> GetResult<T>(Func<IConfigurationSource, string, IConfigurationTelemetry, Func<T, bool>?, bool, ConfigurationResult<T>> selector, Func<T, bool>? validator, bool recordValue)
         {
-            var result = selector(Source, Key, Telemetry, validator, recordValue);
-            if (result.ShouldFallBack && FallbackKey1 is not null)
-            {
-                result = selector(Source, FallbackKey1, Telemetry, validator, recordValue);
-            }
-
-            if (result.ShouldFallBack && FallbackKey2 is not null)
-            {
-                result = selector(Source, FallbackKey2, Telemetry, validator, recordValue);
-            }
-
-            if (result.ShouldFallBack && FallbackKey3 is not null)
-            {
-                result = selector(Source, FallbackKey3, Telemetry, validator, recordValue);
-            }
-
-            return result;
+            var source = Source;
+            var telemetry = Telemetry;
+            return GetResultWithFallback(key => selector(source, key, telemetry, validator, recordValue));
         }
```

Comment:
nit: can probably just simplify these calls. There's a typo on `key` though that definitely _needs_ to be fixed (otherwise you're accessing the constructor parameter)

```csharp
        private ConfigurationResult<T> GetResult<T>(Func<IConfigurationSource, string, IConfigurationTelemetry, Func<T, bool>?, bool, ConfigurationResult<T>> selector, Func<T, bool>? validator, bool recordValue)
            => GetResultWithFallback(key => selector(Source, Key, Telemetry, validator, recordValue));
```

Reply:
mhh I'm not sure I understand, if I pass Key in the selector, I'm not going to be able to switch on aliases in GetResultWithFallback() where we do
```
 private ConfigurationResult<T> GetResultWithFallback<T>(Func<string, ConfigurationResult<T>> selector)
        {
            var result = selector(Key);
            if (!result.ShouldFallBack)
            {
                return result;
            }

            var aliases = _providedAliases ?? ConfigKeyAliasesSwitcher.GetAliases(Key);

            foreach (var alias in aliases)
            {
                result = selector(alias);
                if (!result.ShouldFallBack)
                {
                    break;
                }
            }

            return result;
        }
```

Reply:
Tried simplifying some calls, or removing some passthrough functions in 667cda06519f3789e8b097604ebda27ef35ba3b8